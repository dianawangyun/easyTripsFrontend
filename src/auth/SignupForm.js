import React, { useState } from "react";
import { useHistory } from "react-router-dom";

function SignupForm({ signup }) {
    const INIT_VALUES = {
        username: "",
        password: "",
        firstName: "",
        lastName: "",
        email: "",
    };

    const history = useHistory();
    const [formData, setFormData] = useState(INIT_VALUES);
    const [formErrors, setFormErrors] = useState([]);

    async function handleSubmit(e) {
        e.preventDefault();
        let res = await signup(formData);
        if (res.success) {
            history.push("/home");
        } else {
            console.log(res.errs);
            setFormErrors(res.errs.map((e) => e.replace("instance.", "")));
        }
    }

    function handleChange(e) {
        const { name, value } = e.target;
        setFormData((data) => ({ ...data, [name]: value }));
    }

    return (
        <div className="container">
            <h2 className="mt-5 mb-4 text-center">Sign Up</h2>
            <div className="row d-flex justify-content-center">
                <div className="col-md-6">
                    <form onSubmit={handleSubmit}>
                        <div className="form-group">
                            <label htmlFor="username" className="mr-2">
                                Username:
                            </label>
                            <input
                                type="text"
                                name="username"
                                id="username"
                                className="form-control"
                                value={formData.username}
                                placeholder="<25 characters"
                                onChange={handleChange}
                                required
                            />
                        </div>
                        <div className="form-group">
                            <label htmlFor="password" className="mr-2">
                                Password:
                            </label>
                            <input
                                type="password"
                                name="password"
                                id="password"
                                className="form-control"
                                value={formData.password}
                                onChange={handleChange}
                                placeholder="6-20 characters"
                                required
                            />
                        </div>
                        <div className="form-group">
                            <label htmlFor="firstname" className="mr-2">
                                First Name:
                            </label>
                            <input
                                type="text"
                                name="firstName"
                                id="firstname"
                                className="form-control"
                                value={formData.firstName}
                                onChange={handleChange}
                                required
                            />
                        </div>
                        <div className="form-group">
                            <label htmlFor="lastName" className="mr-2">
                                Last Name:
                            </label>
                            <input
                                type="text"
                                name="lastName"
                                id="lastname"
                                className="form-control"
                                value={formData.lastName}
                                onChange={handleChange}
                                required
                            />
                        </div>
                        <div className="form-group">
                            <label htmlFor="email" className="mr-2">
                                Email:
                            </label>
                            <input
                                type="text"
                                name="email"
                                id="email"
                                className="form-control"
                                value={formData.email}
                                onChange={handleChange}
                                required
                            />
                        </div>
                        {formErrors.length ? (
                            <ul>
                                {formErrors.map((e) => (
                                    <li className="text-danger">{e}</li>
                                ))}
                            </ul>
                        ) : null}
                        <div className="text-center">
                            <button
                                className="btn btn-success mb-5"
                                type="submit"
                            >
                                Sign Up
                            </button>
                        </div>
                    </form>
                </div>
            </div>
        </div>
    );
}

export default SignupForm;
