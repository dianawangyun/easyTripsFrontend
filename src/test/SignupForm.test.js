import React from "react";
import { render, fireEvent } from "@testing-library/react";
import SignupForm from "../auth/SignupForm";
import { MemoryRouter } from "react-router";

it("renders without crashing", function () {
    render(<SignupForm />);
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <SignupForm />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("should show signup form", function () {
    const { queryByText } = render(<SignupForm />);
    expect(queryByText("First Name:")).toBeInTheDocument();
});
