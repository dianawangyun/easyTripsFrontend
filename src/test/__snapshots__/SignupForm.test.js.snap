// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2
      class="mt-5 mb-4 text-center"
    >
      Sign Up
    </h2>
    <div
      class="row d-flex justify-content-center"
    >
      <div
        class="col-md-6"
      >
        <form>
          <div
            class="form-group"
          >
            <label
              class="mr-2"
              for="username"
            >
              Username:
            </label>
            <input
              class="form-control"
              id="username"
              name="username"
              placeholder="<25 characters"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="mr-2"
              for="password"
            >
              Password:
            </label>
            <input
              class="form-control"
              id="password"
              name="password"
              placeholder="6-20 characters"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="mr-2"
              for="firstname"
            >
              First Name:
            </label>
            <input
              class="form-control"
              id="firstname"
              name="firstName"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="mr-2"
              for="lastName"
            >
              Last Name:
            </label>
            <input
              class="form-control"
              id="lastname"
              name="lastName"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="mr-2"
              for="email"
            >
              Email:
            </label>
            <input
              class="form-control"
              id="email"
              name="email"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="text-center"
          >
            <button
              class="btn btn-success mb-5"
              type="submit"
            >
              Sign Up
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
